src/Features/Core/Portable/Completion/Providers/SymbolCompletionItem.cs

```diff
@@ -82,7 +82,7 @@ public static string EncodeSymbols(IReadOnlyList<ISymbol> symbols)
 
         public static string EncodeSymbol(ISymbol symbol)
         {
-            return SymbolKey.ToString(symbol);
+            return SymbolKey.CreateString(symbol);
```

Comment:
Some member were renamed with the goal of SymbolKey being made public.  We have rules about overloads in a public API that SymbolKey was currently violating, so this brings us into compliance.

Reply:
Note: for making things public, there's an open question i have if we even have the SymbolKey struct, or if this is just a static class with static methods to go to/from strings.

Reply:
I'm fine either way. But if we ended up keeping the struct, maybe we should consider replacing string with SymbolKey from the surface area of SymbolKey (as string would be an implementation detail)

---
In reply to: [303146032](https://github.com/dotnet/roslyn/pull/37140#discussion_r303146032) [](ancestors = 303146032)

Reply:
So 'string' is **def** part of the surface area here.  Because a prime usage of SymbolKey is to be able to serialize/persist/rehydrate from disk/indices/etc.  

Now, there is an open question for me if there's actual value in the SymblKey type as something you can make *instances* of.   I think we may not want that, and we may only want to have static helpers to go from `symbol -> string` and `string -> symbol`.  

As part of making this a public APi we should definitely decide on all that.  However, i'd like to separate out hte public-api versus performance PRs as they're orthogonal.  Thanks!

Reply:
ok, i retract the part about wanting to get rid of being able to have **instances** of symbol key.  Instances are still very useful because we can strongly type APIs and there are interesting other things in the API (like `IEqualityComparer<SymbolKey>`).  If we get rid of the sturct, we'd have to have these all be strings, and i think that would make the API much more difficult to use (since random strings would be passed in easily and unexpectedly).

Reply:
I guess I'm just not clear about the surface area when we make this public. OK, I will wait for that change and restrain myself from commenting on this topic in this PR :)

Reply:
Oh, I just realized that you have made things not supposed be made public later `internal` already.

Reply:
Yeah, i'm not totally certain either!  I think there will be some very good discussions at that point about the right shape of htings.